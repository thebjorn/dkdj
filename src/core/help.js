
import dk from "../dk-obj";


/**
 * Create help nodes for all nodes with class="help" under ``item``.
 * 
 * This is automatically called for body on load, but might need to be called explicitly
 * on any dom that is loaded later.
 * 
 * @param item
 */
export function help(item) {
    dk.$(item).find('.help > .help-text').each(function () {
        let $help = dk.$(this).parent();
        let helptext = dk.$(this);
        // $help.css('position', 'relative');
        let trigger = dk.$('<div class="help-trigger"><a><dk-icon value="question-circle"/></a></div>');
        $help.prepend(trigger);
        
        // .popover comes from bootstrap and is thus difficult to unit test...
        $help.find('.help-trigger > a').addClass('xbtn').attr('tabindex', 10).popover({
            html: true,
            title: helptext.attr('title') || 'Hjelp..',
            container: 'body',
            trigger: 'focus',
            toggle: 'popover',
            placement: helptext.attr('placement') || 'auto left',
            content: helptext.html()
        });
    });
}
