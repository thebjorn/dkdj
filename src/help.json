{
  "Class": "the dk root class",
  "counter": `Return a unique counter value.
     The optional 'name' parameter, will create a label
     'name' <counter>, where the counter starts from startval
     (default 1).`
}
