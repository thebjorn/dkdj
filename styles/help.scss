
// interactive help..

.help {
    position: relative;

    dk-icon { color: blue; }

    > .help-trigger {
        position: absolute;
        width: 20px;
        left: -20px;
        text-align: right;
        top: 0;
    }
}
