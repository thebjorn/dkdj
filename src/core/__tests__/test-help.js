import $ from 'jquery';
import utidy from "../../browser/dk-html";
import page from "../../widgetcore/dk-page";
import {help} from "../help";


test("test-help", () => {
    document.body.innerHTML = `
        <div id="work">
            <h1 class="help">Hello
                <div title="help-title" class="help-text">world</div>
            </h1>
        </div>
    `;
    const work = $('#work');
    page.initialize(document);
    let callargs = null;
    $.fn.popover = function (...args) {
        callargs = args[0];
    };
    
    help('#work');
    
    console.log(work.html());
    console.log('CALLARGS', callargs);
    
    expect(callargs.title).toEqual('help-title');
    expect(callargs.trigger).toEqual('focus');
    expect(callargs.content).toEqual('world');
    
    
    expect(work.html()).toMatchSnapshot();
});
