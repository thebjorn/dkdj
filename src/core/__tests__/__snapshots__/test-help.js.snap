// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-help 1`] = `
"
            <h1 class=\\"help\\"><div class=\\"help-trigger\\"><a class=\\"xbtn\\" tabindex=\\"10\\"><dk-icon value=\\"question-circle\\"></dk-icon></a></div>Hello
                <div title=\\"help-title\\" class=\\"help-text\\">world</div>
            </h1>
        "
`;
